tests/unit/moduleapi/moduleconfigs.tcl

```diff
@@ -182,5 +188,44 @@ start_server {tags {"modules"}} {
         # Ensure configs we rewrote are no longer present
         assert_equal [r config get moduleconfigs.*] ""
     }
+    test {startup moduleconfigs} {
+        # No loadmodule directive
+        set nomodload [start_server [list overrides [list moduleconfigs.string "hello"]]]
+        wait_for_condition 100 50 {
+            ! [is_alive $nomodload]
+        } else {
+            fail "startup should've failed with no load and module configs supplied"
+        }
+        set stdout [dict get $nomodload stdout]
+        assert_equal [count_message_lines $stdout "Module Configuration detected without loadmodule directive or no ApplyConfig call: aborting"] 1
+
+        # Bad config value
+        set badconfig [start_server [list overrides [list loadmodule "$testmodule" moduleconfigs.string "rejectisfreed"]]]
+        wait_for_condition 100 50 {
+            ! [is_alive $badconfig]
+        } else {
+            fail "startup with bad moduleconfigs should've failed"
+        }
+        set stdout [dict get $badconfig stdout]
+        assert_equal [count_message_lines $stdout "Issue during loading of configuration moduleconfigs.string"] 1
```

Comment:
let's match against the actual error message (`Cannot set string to 'rejectisfreed'`)